base/scripts/code.player.cs

```diff
@@ -244,35 +246,47 @@
         }
         
         //handle damage events here - ignoring all team dealt damage / self damage
-        if (%teamDamageOccured == 0) {
-            
+        if ($Collector::DamageEnabled && %teamDamageOccured == 0) {
             
+            if(%shooterClient == %damagedClient) { return; }
+
             %newValue = (%value * 150);
-            %newRemainder = (%newValue - floor(%newValue));
             
-            if (%newRemainder >= 0.5) {
-                %newValue = floor((%newValue + 1));
+            if (%spillOver < 0) {
+                //do nothing
             }
             else {
-                %newValue = floor(%newValue);
+                %newValue = (%newValue - (%spillOver * 150));
             }
             
+            $DmgTracker::DamageOut[%shooterClient] += %newValue;
+            $DmgTracker::DamageIn[%damagedClient] += %newValue;
+              
+        }
+        
+        else if ($Collector::DamageEnabled && %teamDamageOccured == 1) {
+            
+            %newValue = (%value * 150);
+            
             if (%spillOver < 0) {
-                //MessageAll(0, "Damage Dealt: " @ %newValue);
+                //do nothing
             }
             else {
-                %newValue = (1+(%newValue - floor(%spillOver * 150)));
-                //MessageAll(0, "New Damage Dealt: " @ %newValue);
+                %newValue = (%newValue - (%spillOver * 150));
             }
             
-            if ($Collector::DamageEnabled) {
-                
-                $DmgTracker::DamageOut[%shooterClient] += %newValue;
-                $DmgTracker::DamageIn[%damagedClient] += %newValue;
-                
-                //zadmin::ActiveMessage::All( DamageDealt, %shooterClient, %damagedClient, %newValue );
-            }
+            $DmgTracker::TeamDamageOut[%shooterClient] += %newValue;
+            $DmgTracker::TeamDamageIn[%damagedClient] += %newValue;
+
         }
+        else { }
+    }
+    
+    if ( (%type == $BulletDamageType) && (%teamDamageOccured == 0) ) {
+        
+            %shooterTeam = Client::getTeam(%shooterClient);
+            Client::sendMessage(%shooterClient,0,"~whit.wav");
+            zadmin::ActiveMessage::All( HitCG, %shooterClient, %shooterTeam );

```

Comment:
could you send `isFriendly` or something instead? otherwise the client has to look up their team and compare to the shooterTeam, all they really care about is whether its friendly fire or not

Reply:
yes you are correct, team is not needed here, will remove